File: python/example_traditional.py
#!/usr/bin/env python

import ROOT

ROOT.gROOT.SetBatch(True)

n = ROOT.TChain("DataTree")
n.Add("data.root")
entries = n.GetEntries()
print "Entries: ", entries

gauss_h = ROOT.TH1F("gauss", "Gaussian distribution", 100, -5., 5.)
uniform_h = ROOT.TH1I("uniform", "Uniform distribution", 100, 0, 101)

for entry in range(entries):
    n.GetEntry(entry)
    gauss_h.Fill(n.gauss_0)
    uniform_h.Fill(n.uniform_0)

canvas = ROOT.TCanvas("c", "c", 400, 400)
gauss_h.Draw()
canvas.SaveAs("gauss.traditional_py.pdf")
canvas.Clear()
uniform_h.Draw()
canvas.SaveAs("uniform.traditional_py.pdf")

File: src/example_traditional.cpp
#include <iostream>
#include <vector>

#include <TROOT.h>
#include <TChain.h>
#include <TH1F.h>
#include <TH1I.h>
#include <TCanvas.h>

int main(int argc, const char *argv[]) {
  gROOT->SetBatch(kTRUE);

  TChain n("DataTree");
  n.Add("data.root");
  ULong64_t entries = n.GetEntries();
  std::cout << "Entries: " << entries << std::endl;

  Double_t gauss;
  Int_t uniform;
  n.SetBranchAddress("gauss_0", &gauss);
  n.SetBranchAddress("uniform_0", &uniform);

  TH1F gauss_h("gauss", "Gaussian distribution", 100, -5., 5.);
  TH1I uniform_h("uniform", "Uniform distribution", 101, 0, 101);
  for (ULong64_t entry = 0; entry < entries; ++entry) {
    n.GetEntry(entry);
    gauss_h.Fill(gauss);
    uniform_h.Fill(uniform);
  }

  TCanvas c("c", "c", 400, 400);
  gauss_h.Draw();
  c.SaveAs("gauss.traditional_cpp.pdf");
  c.Clear();
  uniform_h.Draw();
  c.SaveAs("uniform.traditional_cpp.pdf");

  return 0;
}
